apps/cli/src/commands/live/ledgerSync.ts

```diff
@@ -0,0 +1,160 @@
+    },
+    {
+      name: "rootId",
+      type: String,
+      desc: "The immutable id of the Tree root retrieved from getKeyRingTree result",
+    },
+    {
+      name: "walletSyncEncryptionKey",
+      type: String,
+      desc: "The secret used to encrypt/decrypt the wallet sync data retrieved from getKeyRingTree result",
+    },
+    {
+      name: "applicationPath",
+      type: String,
+      desc: "privatekey for Ledger Key Ring Protocol Tree from initMemberCredentials result",
+    },
+    {
+      name: "data",
+      type: String,
+      desc: "data to be pushed to Ledger Sync",
+    },
+    {
+      name: "version",
+      type: Number,
+      desc: "version of the data",
+    },
+    {
+      name: "applicationId",
+      type: Number,
+      default: 16,
+      desc: "application identifier",
+    },
+    {
+      name: "name",
+      type: String,
+      default: "CLI",
+      desc: "name of the instance",
+    },
+    {
+      name: "apiBaseUrl",
+      type: String,
+      default: getEnv("TRUSTCHAIN_API_STAGING"),
+      desc: "api base url for Ledger Key Ring Protocol",
+    },
+    {
+      name: "cloudSyncApiBaseUrl",
+      type: String,
+      default: getEnv("CLOUD_SYNC_API_STAGING"),
+      desc: "api base url for Cloud Sync",
+    },
+  ],
+  job: ({
+    push,
+    pull,
+    pubKey,
+    privateKey,
+    rootId,
+    walletSyncEncryptionKey,
+    applicationPath,
+    data,
+    version,
+    applicationId = 16,
+    name = "CLI",
+    apiBaseUrl = getEnv("TRUSTCHAIN_API_STAGING"),
+    cloudSyncApiBaseUrl = getEnv("CLOUD_SYNC_API_STAGING"),
+  }: Partial<{
+    push: boolean;
+    pull: boolean;
+    pubKey: string;
+    privateKey: string;
+    rootId: string;
+    walletSyncEncryptionKey: string;
+    applicationPath: string;
+    data: string;
+    version: number;
+    applicationId: number;
+    name: string;
+    apiBaseUrl: string;
+    cloudSyncApiBaseUrl: string;
+  }>) => {
+    if (!applicationId) return "applicationId is required";
+    if (!name) return "name is required";
+    if (!apiBaseUrl) return "apiBaseUrl is required";
+    if (!pubKey || !privateKey) return "pubKey and privateKey are required";
+    if (!rootId) return "pubKey and privateKey are required";
+    if (!walletSyncEncryptionKey) return "walletSyncEncryptionKey is required";
+    if (!applicationPath) return "applicationPath is required";
+
+    const context = {
+      applicationId,
+      name,
+      apiBaseUrl,
+    };
+    const ledgerKeyRingProtocolSDK = getSdk(false, context, withDevice);
+
+    const cloudSyncSDK = new CloudSyncSDK({
+      apiBaseUrl: cloudSyncApiBaseUrl,
+      slug: liveSlug,
+      schema: walletsync.schema,
+      trustchainSdk: ledgerKeyRingProtocolSDK,
+      getCurrentVersion: () => version || 1,
+      saveNewUpdate: async (event: UpdateEvent<LiveData>) => {
+        console.log(event);
+      },
+    });
+
+    if (push) {
+      if (!data) return "data is required";
```

Comment:
question: why this check is not made as a return-early with those between line:112->118 ? it would avoid `getSdk` and `cloudSyncSDK` instantiation for nothing. am i wrong ?

Reply:
Good point, thanks!